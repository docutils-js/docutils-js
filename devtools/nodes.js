#!/usr/bin/env node
const recast = require('recast');
const j = require('jscodeshift');
const fs = require('fs');
const path = require('path');
const PropTypes = require('prop-types');
const nodes = require('../lib/nodes');
const nodesSrc = path.join(__dirname, '../src/nodes.js');
// const nodesJs = fs.createReadStream(nodesSrc);
const source = fs.readFileSync(nodesSrc, { encoding: 'utf-8' });
const file = recast.parse(source);

const superClasses = [];
const leafClasses = [];
let classes = ['Node'];
const r = j(source);
const paths = r.find(j.ClassDeclaration, p => p.superClass).paths();
const nodesInfo = {};
let curNodeInfo;
const flatNodesInfo = {};
/* Each iteration we process the Nth level of the class hierarchy, starting from the root class Node */
while (classes.length) {
    const nextClasses = [];
    classes.forEach((superClass) => {
        curNodeInfo = nodesInfo[superClass];
        const subClasses = paths.filter(p => p.value.superClass.name === superClass).map(p => p.value.id.name);
        const myInfo = {};
        myInfo.leaf = subClasses.length === 0;
        const args = [];
        if (subClasses.length) {
            myInfo.subClasses = subClasses;
            superClasses.push(superClass);
        } else {
            leafClasses.push(superClass);
            const params = nodes[superClass].params;
            if (params) {
                if (Array.isArray(params)) {
                    params.forEach(param => {
                        if (param == nodes.Params.ElementContent) {
                            args.push('Element Content');
                        } else if (param == nodes.Params.RawSource) {
                            args.push('Raw Source');
                        }
                    });
                }
                ;
            }
            //PropTypes.checkPropTypes(nodes[superClass].propTypes, {});
            console.log(superClass);
        }
        try {
            const node = new nodes[superClass](...args);
            myInfo.classTypes = node.classTypes.map(c => c.name);
        } catch(err) {
            myInfo.error = err.message;
        }
//            if(!node.test()) {
//                throw new Error();;
//            }
            myInfo.name = superClass;
            flatNodesInfo[myInfo.name] = myInfo;

        nextClasses.splice(nextClasses.length, 0, ...subClasses);
        console.log(myInfo);
    });
    classes = nextClasses;
}
fs.writeFileSync('nodes.json', JSON.stringify(flatNodesInfo));

console.log(JSON.stringify(flatNodesInfo));
const transBody = j.classBody([]);
const file2 = recast.parse('import BaseWriter from \'../Writer\';\n'
                           + 'import { GenericNodeVisitor } from \'../nodes\';\n');
const classDecl = j.classDeclaration(j.identifier('Translator'), transBody, j.identifier('GenericNodeVisitor'));
file2.program.body.push(classDecl);
const f = file2;
// j.program([j.importDeclaration([j.importNamespaceSpecifier(j.identifier('BaseWriter'))],
//    j.literal('../Writer'), ]));
// console.log(Object.keys(api).filter(x => 1 || x.toLowerCase().indexOf('recast')!== -1));
const methods = [];
leafClasses.forEach((class_) => {
    const params = [j.identifier('node')];

    const m = j.methodDefinition('method', j.identifier(`visit_${class_}`),
                                 j.functionExpression(null, [j.identifier('node')], j.blockStatement([])));
    const eslintHint = j.commentBlock(' eslint-disable-next-line camelcase,no-unused-vars,class-methods-use-this ');

    m.comments = [eslintHint];
    const depart = j.methodDefinition('method', j.identifier(`depart_${class_}`),
                                      j.functionExpression(null, [j.identifier('node')], j.blockStatement([])));
    depart.comments = [eslintHint];
    // console.log(m);
    transBody.body.push(m, depart);
    // methods.push(m.toString());
});
fs.createWriteStream('code.js').write(recast.print(f).code);
