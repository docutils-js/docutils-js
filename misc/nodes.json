{
   "footnote_reference" : {
      "leaf" : true,
      "name" : "footnote_reference",
      "classTypes" : [
         "General",
         "Inline",
         "Referential"
      ]
   },
   "address" : {
      "name" : "address",
      "classTypes" : [
         "Bibliographic"
      ],
      "leaf" : true
   },
   "literal_block" : {
      "leaf" : true,
      "name" : "literal_block",
      "classTypes" : [
         "General"
      ]
   },
   "block_quote" : {
      "leaf" : true,
      "name" : "block_quote",
      "classTypes" : [
         "General"
      ]
   },
   "field_list" : {
      "leaf" : true,
      "name" : "field_list",
      "classTypes" : [
         "Sequential"
      ]
   },
   "reference" : {
      "leaf" : true,
      "name" : "reference",
      "classTypes" : [
         "General",
         "Inline",
         "Referential"
      ]
   },
   "compound" : {
      "leaf" : true,
      "name" : "compound",
      "classTypes" : [
         "General"
      ]
   },
   "citation_reference" : {
      "name" : "citation_reference",
      "classTypes" : [
         "General",
         "Inline",
         "Referential"
      ],
      "leaf" : true
   },
   "classifier" : {
      "leaf" : true,
      "classTypes" : [
         "Part"
      ],
      "name" : "classifier"
   },
   "substitution_reference" : {
      "name" : "substitution_reference",
      "classTypes" : [
         "Inline"
      ],
      "leaf" : true
   },
   "option_list_item" : {
      "classTypes" : [
         "Part"
      ],
      "name" : "option_list_item",
      "leaf" : true
   },
   "note" : {
      "leaf" : true,
      "classTypes" : [
         "Admonition"
      ],
      "name" : "note"
   },
   "important" : {
      "leaf" : true,
      "classTypes" : [
         "Admonition"
      ],
      "name" : "important"
   },
   "comment" : {
      "leaf" : true,
      "name" : "comment",
      "classTypes" : [
         "Special",
         "Invisible",
         "Inline",
         "Targetable"
      ]
   },
   "bullet_list" : {
      "classTypes" : [
         "Sequential"
      ],
      "name" : "bullet_list",
      "leaf" : true
   },
   "table" : {
      "classTypes" : [
         "General"
      ],
      "name" : "table",
      "leaf" : true
   },
   "container" : {
      "leaf" : true,
      "name" : "container",
      "classTypes" : [
         "General"
      ]
   },
   "version" : {
      "classTypes" : [
         "Bibliographic"
      ],
      "name" : "version",
      "leaf" : true
   },
   "hint" : {
      "leaf" : true,
      "classTypes" : [
         "Admonition"
      ],
      "name" : "hint"
   },
   "revision" : {
      "classTypes" : [
         "Bibliographic"
      ],
      "name" : "revision",
      "leaf" : true
   },
   "citation" : {
      "classTypes" : [
         "General",
         "BackLinkable",
         "Labeled",
         "Targetable"
      ],
      "name" : "citation",
      "leaf" : true
   },
   "subtitle" : {
      "name" : "subtitle",
      "classTypes" : [
         "Titular",
         "PreBibliographic"
      ],
      "leaf" : true
   },
   "line" : {
      "leaf" : true,
      "name" : "line",
      "classTypes" : [
         "Part"
      ]
   },
   "emphasis" : {
      "name" : "emphasis",
      "classTypes" : [
         "Inline"
      ],
      "leaf" : true
   },
   "footer" : {
      "classTypes" : [
         "Decorative"
      ],
      "name" : "footer",
      "leaf" : true
   },
   "definition_list" : {
      "classTypes" : [
         "Sequential"
      ],
      "name" : "definition_list",
      "leaf" : true
   },
   "footnote" : {
      "leaf" : true,
      "name" : "footnote",
      "classTypes" : [
         "General",
         "BackLinkable",
         "Labeled",
         "Targetable"
      ]
   },
   "author" : {
      "name" : "author",
      "classTypes" : [
         "Bibliographic"
      ],
      "leaf" : true
   },
   "row" : {
      "name" : "row",
      "classTypes" : [
         "Part"
      ],
      "leaf" : true
   },
   "figure" : {
      "leaf" : true,
      "classTypes" : [
         "General"
      ],
      "name" : "figure"
   },
   "docinfo" : {
      "classTypes" : [
         "Bibliographic"
      ],
      "name" : "docinfo",
      "leaf" : true
   },
   "copyright" : {
      "classTypes" : [
         "Bibliographic"
      ],
      "name" : "copyright",
      "leaf" : true
   },
   "title_reference" : {
      "classTypes" : [
         "Inline"
      ],
      "name" : "title_reference",
      "leaf" : true
   },
   "math_block" : {
      "leaf" : true,
      "name" : "math_block",
      "classTypes" : [
         "General"
      ]
   },
   "acronym" : {
      "leaf" : true,
      "classTypes" : [
         "Inline"
      ],
      "name" : "acronym"
   },
   "superscript" : {
      "classTypes" : [
         "Inline"
      ],
      "name" : "superscript",
      "leaf" : true
   },
   "system_message" : {
      "classTypes" : [
         "Special",
         "BackLinkable",
         "PreBibliographic"
      ],
      "name" : "system_message",
      "leaf" : true
   },
   "substitution_definition" : {
      "leaf" : true,
      "classTypes" : [
         "Special",
         "Invisible"
      ],
      "name" : "substitution_definition"
   },
   "error" : {
      "leaf" : true,
      "name" : "error",
      "classTypes" : [
         "Admonition"
      ]
   },
   "option_group" : {
      "leaf" : true,
      "classTypes" : [
         "Part"
      ],
      "name" : "option_group"
   },
   "danger" : {
      "leaf" : true,
      "name" : "danger",
      "classTypes" : [
         "Admonition"
      ]
   },
   "target" : {
      "name" : "target",
      "classTypes" : [
         "Special",
         "Invisible",
         "Inline",
         "Targetable"
      ],
      "leaf" : true
   },
   "attention" : {
      "leaf" : true,
      "name" : "attention",
      "classTypes" : [
         "Admonition"
      ]
   },
   "admonition" : {
      "name" : "admonition",
      "classTypes" : [
         "Admonition"
      ],
      "leaf" : true
   },
   "topic" : {
      "name" : "topic",
      "classTypes" : [
         "Structural"
      ],
      "leaf" : true
   },
   "paragraph" : {
      "leaf" : true,
      "classTypes" : [
         "General"
      ],
      "name" : "paragraph"
   },
   "rubric" : {
      "leaf" : true,
      "name" : "rubric",
      "classTypes" : [
         "Titular"
      ]
   },
   "label" : {
      "leaf" : true,
      "name" : "label",
      "classTypes" : [
         "Part"
      ]
   },
   "Element" : {
      "subClasses" : [
         "header",
         "footer",
         "decoration",
         "TextElement",
         "document",
         "docinfo",
         "authors",
         "section",
         "topic",
         "sidebar",
         "transition",
         "compound",
         "container",
         "bullet_list",
         "enumerated_list",
         "list_item",
         "definition_list",
         "definition_list_item",
         "definition",
         "field_list",
         "field",
         "field_body",
         "option",
         "option_group",
         "option_list",
         "option_list_item",
         "description",
         "line_block",
         "block_quote",
         "attention",
         "caution",
         "danger",
         "error",
         "important",
         "note",
         "tip",
         "hint",
         "warning",
         "admonition",
         "footnote",
         "citation",
         "figure",
         "legend",
         "table",
         "tgroup",
         "colspec",
         "thead",
         "tbody",
         "row",
         "entry",
         "system_message",
         "pending",
         "image"
      ],
      "name" : "Element",
      "classTypes" : [],
      "leaf" : false
   },
   "subscript" : {
      "classTypes" : [
         "Inline"
      ],
      "name" : "subscript",
      "leaf" : true
   },
   "document" : {
      "leaf" : true,
      "classTypes" : [
         "Root",
         "Structural"
      ],
      "name" : "document"
   },
   "field_name" : {
      "classTypes" : [
         "Part"
      ],
      "name" : "field_name",
      "leaf" : true
   },
   "sidebar" : {
      "name" : "sidebar",
      "classTypes" : [
         "Structural"
      ],
      "leaf" : true
   },
   "authors" : {
      "leaf" : true,
      "name" : "authors",
      "classTypes" : [
         "Bibliographic"
      ]
   },
   "tip" : {
      "leaf" : true,
      "classTypes" : [
         "Admonition"
      ],
      "name" : "tip"
   },
   "attribution" : {
      "name" : "attribution",
      "classTypes" : [
         "Part"
      ],
      "leaf" : true
   },
   "generated" : {
      "leaf" : true,
      "name" : "generated",
      "classTypes" : [
         "Inline"
      ]
   },
   "math" : {
      "leaf" : true,
      "classTypes" : [
         "Inline"
      ],
      "name" : "math"
   },
   "option" : {
      "classTypes" : [
         "Part"
      ],
      "name" : "option",
      "leaf" : true
   },
   "entry" : {
      "classTypes" : [
         "Part"
      ],
      "name" : "entry",
      "leaf" : true
   },
   "legend" : {
      "classTypes" : [
         "Part"
      ],
      "name" : "legend",
      "leaf" : true
   },
   "warning" : {
      "leaf" : true,
      "name" : "warning",
      "classTypes" : [
         "Admonition"
      ]
   },
   "description" : {
      "leaf" : true,
      "classTypes" : [
         "Part"
      ],
      "name" : "description"
   },
   "date" : {
      "name" : "date",
      "classTypes" : [
         "Bibliographic"
      ],
      "leaf" : true
   },
   "list_item" : {
      "name" : "list_item",
      "classTypes" : [
         "Part"
      ],
      "leaf" : true
   },
   "option_argument" : {
      "name" : "option_argument",
      "classTypes" : [
         "Part"
      ],
      "leaf" : true
   },
   "tgroup" : {
      "name" : "tgroup",
      "classTypes" : [
         "Part"
      ],
      "leaf" : true
   },
   "strong" : {
      "leaf" : true,
      "classTypes" : [
         "Inline"
      ],
      "name" : "strong"
   },
   "literal" : {
      "leaf" : true,
      "name" : "literal",
      "classTypes" : [
         "Inline"
      ]
   },
   "caution" : {
      "leaf" : true,
      "classTypes" : [
         "Admonition"
      ],
      "name" : "caution"
   },
   "section" : {
      "leaf" : true,
      "classTypes" : [
         "Structural"
      ],
      "name" : "section"
   },
   "image" : {
      "leaf" : true,
      "name" : "image",
      "classTypes" : [
         "General",
         "Inline"
      ]
   },
   "header" : {
      "leaf" : true,
      "name" : "header",
      "classTypes" : [
         "Decorative"
      ]
   },
   "definition_list_item" : {
      "leaf" : true,
      "classTypes" : [
         "Part"
      ],
      "name" : "definition_list_item"
   },
   "decoration" : {
      "classTypes" : [
         "Decorative"
      ],
      "name" : "decoration",
      "leaf" : true
   },
   "Node" : {
      "subClasses" : [
         "Element",
         "Text"
      ],
      "name" : "Node",
      "classTypes" : [],
      "leaf" : false
   },
   "field" : {
      "name" : "field",
      "classTypes" : [
         "Part"
      ],
      "leaf" : true
   },
   "option_string" : {
      "leaf" : true,
      "classTypes" : [
         "Part"
      ],
      "name" : "option_string"
   },
   "field_body" : {
      "leaf" : true,
      "classTypes" : [
         "Part"
      ],
      "name" : "field_body"
   },
   "raw" : {
      "name" : "raw",
      "classTypes" : [
         "Special",
         "Inline",
         "PreBibliographic"
      ],
      "leaf" : true
   },
   "doctest_block" : {
      "leaf" : true,
      "classTypes" : [
         "General"
      ],
      "name" : "doctest_block"
   },
   "inline" : {
      "name" : "inline",
      "classTypes" : [
         "Inline"
      ],
      "leaf" : true
   },
   "transition" : {
      "leaf" : true,
      "classTypes" : [
         "Structural"
      ],
      "name" : "transition"
   },
   "thead" : {
      "leaf" : true,
      "name" : "thead",
      "classTypes" : [
         "Part"
      ]
   },
   "status" : {
      "classTypes" : [
         "Bibliographic"
      ],
      "name" : "status",
      "leaf" : true
   },
   "pending" : {
      "leaf" : true,
      "classTypes" : [],
      "name" : "pending"
   },
   "colspec" : {
      "leaf" : true,
      "name" : "colspec",
      "classTypes" : [
         "Part"
      ]
   },
   "tbody" : {
      "leaf" : true,
      "classTypes" : [
         "Part"
      ],
      "name" : "tbody"
   },
   "title" : {
      "leaf" : true,
      "name" : "title",
      "classTypes" : [
         "Titular",
         "PreBibliographic"
      ]
   },
   "problematic" : {
      "name" : "problematic",
      "classTypes" : [
         "Inline"
      ],
      "leaf" : true
   },
   "term" : {
      "leaf" : true,
      "classTypes" : [
         "Part"
      ],
      "name" : "term"
   },
   "organization" : {
      "leaf" : true,
      "name" : "organization",
      "classTypes" : [
         "Bibliographic"
      ]
   },
   "TextElement" : {
      "classTypes" : [],
      "name" : "TextElement",
      "subClasses" : [
         "FixedTextElement",
         "title",
         "subtitle",
         "rubric",
         "author",
         "organization",
         "contact",
         "version",
         "revision",
         "status",
         "date",
         "copyright",
         "paragraph",
         "term",
         "classifier",
         "field_name",
         "option_argument",
         "option_string",
         "line",
         "attribution",
         "substitution_definition",
         "target",
         "label",
         "caption",
         "emphasis",
         "strong",
         "literal",
         "reference",
         "footnote_reference",
         "citation_reference",
         "substitution_reference",
         "title_reference",
         "abbreviation",
         "acronym",
         "superscript",
         "subscript",
         "math",
         "inline",
         "problematic",
         "generated"
      ],
      "leaf" : false
   },
   "caption" : {
      "classTypes" : [
         "Part"
      ],
      "name" : "caption",
      "leaf" : true
   },
   "option_list" : {
      "classTypes" : [
         "Sequential"
      ],
      "name" : "option_list",
      "leaf" : true
   },
   "abbreviation" : {
      "name" : "abbreviation",
      "classTypes" : [
         "Inline"
      ],
      "leaf" : true
   },
   "Text" : {
      "classTypes" : [],
      "name" : "Text",
      "leaf" : true
   },
   "contact" : {
      "classTypes" : [
         "Bibliographic"
      ],
      "name" : "contact",
      "leaf" : true
   },
   "FixedTextElement" : {
      "leaf" : false,
      "subClasses" : [
         "address",
         "literal_block",
         "doctest_block",
         "math_block",
         "comment",
         "raw"
      ],
      "name" : "FixedTextElement",
      "classTypes" : []
   },
   "line_block" : {
      "leaf" : true,
      "classTypes" : [
         "General"
      ],
      "name" : "line_block"
   },
   "definition" : {
      "leaf" : true,
      "name" : "definition",
      "classTypes" : [
         "Part"
      ]
   },
   "enumerated_list" : {
      "classTypes" : [
         "Sequential"
      ],
      "name" : "enumerated_list",
      "leaf" : true
   }
}
