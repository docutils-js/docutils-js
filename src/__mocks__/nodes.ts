export const mockNoteCitation = jest.fn();
// @ts=ignore
const nodes = jest.requireActual('../nodes');

const mockDocument = jest.fn().mockImplementation(() => {
    return { noteCitation: mockNoteCitation };
});
nodes.document = mockDocument;

