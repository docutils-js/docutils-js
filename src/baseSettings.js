const default_error_encoding_error_handler = 'backslashreplace';

export default {
    tocBacklinks: 'entry',
    footnoteBacklinks: true,
    sectionNumbering: true,
    reportLevel: 2,
    haltLevel: 4,
    exitStatusLevel: 5,
    debug: true,
    traceback: null,
    inputEncodingErrorHandler: 'strict',
    outputEncoding: 'utf-8',
    outputEncodingErrorHandler: 'strict',
    errorEncodingErrorHandler: default_error_encoding_error_handler,
    errorEncoding: 'utf-8',
    languageCode: 'en',
    recordDependencies: null,
    idPrefix: '',
    autoIdPrefix: 'id',
    dumpSettings: null,
    dumpInternals: null,
    dumpTransforms: null,
    dumpPsuedoXml: null,
    exposeInternalAttribute: null,
    strictVisitor: null,
    warningStream: null,
};
