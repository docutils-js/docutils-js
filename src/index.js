export const __version__ = '0.14js';
