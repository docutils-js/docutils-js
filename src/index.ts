/* eslint-disable-next-line import/prefer-default-export */
export { parse } from './parse';
export { StringOutput, StringInput } from './io';
export { StandaloneReader } from './readers/standalone';
export { newDocument } from './newDocument';
import * as nodes from './nodes';
import Writer from './Writer';
import Transform from './Transform';
export { Reader } from './Reader';
export { Publisher } from './Publisher'
export { defaults } from '../gen/defaults';
export { pojoTranslate } from './fn/pojoTranslate';
export { htmlTranslate } from './fn/htmlTranslate';
export { RSTParser } from './parsers/restructuredtext';
export { XMLWriter } from './writers/xml'

/* eslint-disable-next-line @typescript-eslint/no-unused-vars,no-unused-vars */
const __docformat__ = 'reStructuredText';

export const __version__ = '0.14js';

export {
    nodes, Writer, Transform,
};
