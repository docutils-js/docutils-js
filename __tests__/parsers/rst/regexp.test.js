import Inliner from '../../../src/parsers/rst/Inliner';
import { newDocument, newReporter, isIterable } from '../../../src/utils';
import { Element } from '../../../src/nodes';
import baseSettings from '../../../src/baseSettings';

function testInitialInlinerPattern(text, inliner) {
    if(typeof inliner === 'undefined') {
	inliner = setupInliner();
    }
    const result = inliner.patterns.initial[0].exec(text);
    if(!result) {
	return undefined;
    }
    const rr = {}
    inliner.patterns.initial[1].forEach((x, index) => {
	if (x != null) {
            rr[x] = result[index];
	}
    });
    return { groups: rr,
	     input: text,
	     result,
	     pattern: inliner.patterns.initial[0]
	   };
}

function setupInliner() {
    const inliner = new Inliner();
    inliner.initCustomizations({});
    const document = newDocument({}, baseSettings);
    const reporter = newReporter({}, {});
    let language;
    const memo = { document,
		   reporter,
		   language,
		 };
    return inliner;
}

function processRegexp(definition, text) {
    let [fakeTuple, name, prefix, suffix, parts] = definition;
    let prefixNames = [];
    if (Array.isArray(prefix)) {
        prefix.shift();
        const pr = prefix.shift();
        prefixNames = [...prefix];
        prefix = pr;
    }
    if (suffix === undefined) {
        throw new Error();
    }
    let suffixNames = [];
    if (Array.isArray(suffix)) {
        suffix.shift();
        const sr = suffix.shift();
        suffixNames = [...suffix];
        suffix = sr;
    }

    if (!fakeTuple) {
        throw new Error();
    }
    const pi = isIterable(parts);
//    console.log(`buildRegexp(${name} - ${pi})`);
    const partStrings = [];
//    console.log(parts);
    if (parts === undefined) {
        throw new Error();
    }
    const fakeTuple2 = parts.shift();
    const groupNames = [];
    for (const part of parts) {
        const fakeTuple3 = Array.isArray(part) ? part[0] : undefined;
        if (fakeTuple3 === 1) {
            const [regexp, subGroupNames] = buildRegexp(part, null);
            groupNames.push(...subGroupNames);
            partStrings.push(regexp);
        } else if (fakeTuple3 === 2) {
            part.shift();
            const regexp = part.shift();
            partStrings.push(regexp);
                groupNames.push(...part);
            } else {
                partStrings.push(part);
            }
    }
    const orGroup = partStrings.map(x => `(${x})`).join('|');
    const regexp = `${prefix}(${orGroup})${suffix}`;
//    console.log(new RegExp(regexp))
    groupNames.splice(0, 0, ...prefixNames, name);
    groupNames.push(...suffixNames);
//    console.log('groupnames')
//    console.log(groupNames);
//    console.log(`regexp is ${regexp}`);
    if (compile) {
        return [new RegExp(regexp), groupNames, regexp];
    }

    return [regexp, groupNames];
}

test.each([
    ['interpreted with role prefix', ':hello:`test`'],
    ['interpreted with role suffix', '`test`:hello:'],
    ['emphasis surrounded by text', 'hello *and* goodbye'],
])("%s", (a, b) => {

    const results = testInitialInlinerPattern(b);
    expect(results.groups).toMatchSnapshot();
});
