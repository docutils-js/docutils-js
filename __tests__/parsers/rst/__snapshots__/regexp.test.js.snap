// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`emphasis surrounded by text 1`] = `
Object {
  "backquote": undefined,
  "citationlabel": undefined,
  "fnend": undefined,
  "footnotelabel": undefined,
  "initial_inline": "*",
  "refend": undefined,
  "refname": undefined,
  "role": undefined,
  "start": "*",
  "whole": undefined,
}
`;

exports[`interpreted with role prefix 1`] = `
Object {
  "backquote": "\`",
  "citationlabel": undefined,
  "fnend": undefined,
  "footnotelabel": undefined,
  "initial_inline": ":hello:\`",
  "refend": undefined,
  "refname": undefined,
  "role": ":hello:",
  "start": undefined,
  "whole": undefined,
}
`;

exports[`interpreted with role suffix 1`] = `
Object {
  "backquote": "\`",
  "citationlabel": undefined,
  "fnend": undefined,
  "footnotelabel": undefined,
  "initial_inline": "\`",
  "refend": undefined,
  "refname": undefined,
  "role": "",
  "start": undefined,
  "whole": undefined,
}
`;
